mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m

```diff
@@ -84,16 +94,179 @@ - (void)testprocessEvent_eventForLowercaseA_ReturnsQstrActionWithExpectedCharact
     XCTAssert([output isEqualToString:@"\u00C7"], @"Expected capital C cedille (U+00C7)");
 }
 
-//- (void)testprocessEvent_eventForLeftOptionA_ReturnsQstrActionWithExpectedCharacterBasedOnKmx {
-//    KMXFile *kmxFile = [KeymanEngineTestsStaticHelperMethods getKmxFileTestMacEngine];
-//    KMEngine *engine = [[KMEngine alloc] initWithKMX:kmxFile contextBuffer:@""];
-//    NSEvent *event = [NSEvent keyEventWithType:NSEventTypeKeyDown location:NSMakePoint(0, 0) modifierFlags:NSEventModifierFlagControl timestamp:0 windowNumber:0 context:nil characters:@"å" charactersIgnoringModifiers:@"a" isARepeat:NO keyCode:kVK_ANSI_A];
-//    NSArray *actions = [engine processEvent:event];
-//    XCTAssert(actions.count == 1, @"Expected 1 action");
-//    NSDictionary *action = actions[0];
-//    NSString *actionType = [[action allKeys] objectAtIndex:0];
-//    XCTAssert([actionType isEqualToString:Q_STR], @"Expected Q_STR action");
-//    NSString *output = [action objectForKey:actionType];
-//    XCTAssert([output isEqualToString:@"\u00A3"], @"Expected pound sign (U+00A3)");
-//}
++ (void)fillInNamesAndModifiersForAllChiralCombinations {
+    int i = 0;
+    modifiers[i] = LEFT_CTRL_FLAG;
```

Comment:
Given we have a fixed number of combinations, I'd be more comfortable using integer values than an incrementing index here -- it's cleaner to read and understand., e.g. `modifiers[0] = ...; modifiers[1] = ...;`

Reply:
Maybe so. The advantage of using an index that can be incremented is that it's fast for copying and pasting, especially if you ever stick another entry in somewhere besides the end. I really wanted to use a dictionary, but NSDictionary doesn't support non-object types as either keys or values.

Reply:
The disadvantage is that you can overflow bounds without noticing :)